src/main/java/trackitnus/model/task/Task.java

```diff
@@ -15,55 +16,77 @@
  */
 public class Task {
     public static final String TYPE = "T";
-
-    public static final String DATE_MESSAGE_CONSTRAINTS = "Date should be in the format d/MM/yyyy";
+    public static final String DATE_MESSAGE_CONSTRAINTS = "Date should be in the format dd/MM/yyyy or dd/MM/yyyy hh:mm";
     public static final String WEIGHTAGE_MESSAGE_CONSTRAINTS = "Weightage should be in the"
         + " form of a floating point number";
-    public static final String REMARK_MESSAGE_CONSTRAINTS = "Remarks can take any values, and it should not be blank";
-    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("d/MM/yyyy");
+    public static final DateTimeFormatter FORMATTER = DateTimeFormatter.ofPattern("dd/MM/yyyy");
+    public static final String VALIDATION_REGEX = "[^\\s].*";
 
     private final Name name;
     private final LocalDate date;
-    private final Address address;
-    private final Double weightage;
+    private final Code code;
     private final String remark;
+    private boolean isDone;
 
     /**
      * Every field must be present and not null.
      *
      * @param name
      * @param date
-     * @param address
-     * @param weightage
      * @param remark
      */
-    public Task(Name name, LocalDate date, Address address, Double weightage, String remark) {
-        CollectionUtil.requireAllNonNull(name, date, address, weightage, remark);
+    public Task(Name name, LocalDate date, Code code, String remark) {
+        CollectionUtil.requireAllNonNull(name, date);
+        this.name = name;
+        this.date = date;
+        this.code = code;
+        this.remark = remark;
+        this.isDone = false;
+    }
+
+    /**
+     * Overloaded constructor for Task. Every field must be present and not null.
+     *
+     * @param name
+     * @param date
+     * @param remark
+     * @param isComplete
+     */
+    public Task(Name name, LocalDate date, Code code, String remark, boolean isComplete) {
+        CollectionUtil.requireAllNonNull(name, date);
         this.name = name;
         this.date = date;
-        this.address = address;
-        this.weightage = weightage;
+        this.code = code;
         this.remark = remark;
+        this.isDone = isComplete;
```

Comment:
hey this should be isComplete haha. A trick to faster change the name of a variable is to use Find and Replace haha